File: Dymetric_PeriodicFunction.vb
Public Class Dymetric
    Private sine_curve As New PeriodicFunction
    Private do_simulation = False

    ' decide what course of action to take
    Public Sub decideAction(sender As Object, g As Graphics, click_check As Boolean)
        If do_simulation And click_check Then
            ' do animation
            sine_curve.play(sender, g)
            do_simulation = False
        Else
            ' Put ball on screen
            sine_curve.prep(sender, g)
            do_simulation = True
        End If
    End Sub
End Class

File: Dymetric_PeriodicFunction.cs
using System.Windows.Forms;

namespace Dymetric_CS
{
    class Dymetric
    {
        private PeriodicFunction sine_curve;
        private bool do_simulation;

        public Dymetric(int screen_width, int screen_height)
        {
            sine_curve = new PeriodicFunction(screen_width, screen_height);
            do_simulation = false;
        }

        // decide what course of action to take
        public void decideAction(PaintEventArgs e, bool click_check)
        {
            if (do_simulation && click_check)
            {
                // do animation
                sine_curve.inPlay(e);
                do_simulation = false;
            }
            else
            {
                // Put ball on screen
                sine_curve.clearAndDraw(e);
                do_simulation = true;
            }
        }
    }
}
